test/unit/gateways/securion_pay_test.rb

```diff
@@ -63,6 +72,89 @@ def test_successful_purchase
     assert response.test?
   end
 
+  def test_three_ds_v1_object_construction
+    post = {}
+    @options[:three_ds_options] = @three_ds_secure
+    @gateway.send(:add_three_ds, post, @options)
+    assert post[:threeDSecure]
+    ds_data = post[:threeDSecure][:external]
+    ds_options = @options[:three_ds_options]
+    assert_equal ds_options[:version], ds_data[:version]
+    assert_equal ds_options[:cavv], ds_data[:authenticationValue]
+    assert_equal ds_options[:ecommerce_indicator], ds_data[:eci]
+    assert_equal ds_options[:xid], ds_data[:xid]
+    assert_equal nil, ds_data[:ds_transaction_id]
+    assert_equal ds_options[:authentication_response_status], ds_data[:status]
+  end
+
+  def test_three_ds_v2_object_construction
+    post = {}
+    @three_ds_secure[:ds_transaction_id] = 'ODUzNTYzOTcwODU5NzY3Qw=='
+    @three_ds_secure[:version] = '2.2.0'
+    @options[:three_ds_options] = @three_ds_secure
+
+    @gateway.send(:add_three_ds, post, @options)
+
+    assert post[:threeDSecure]
+    ds_data = post[:threeDSecure][:external]
+    ds_options = @options[:three_ds_options]
+
+    assert_equal ds_options[:version], ds_data[:version]
+    assert_equal ds_options[:cavv], ds_data[:authenticationValue]
+    assert_equal ds_options[:ecommerce_indicator], ds_data[:eci]
+    assert_equal nil, ds_data[:xid]
+    assert_equal ds_options[:ds_transaction_id], ds_data[:dsTransactionId]
+    assert_equal ds_options[:authentication_response_status], ds_data[:status]
+  end
+
+  def test_three_ds_version_validation
+    post = {}
+    @options[:three_ds_options] = @three_ds_secure
+    @gateway.send(:add_three_ds, post, @options)
+    resp = @gateway.send(:validate_three_ds_params, @three_ds_secure)
+
+    assert_equal nil, resp
+    post[:threeDSecure][:external][:version] = '4.0'
+    resp = @gateway.send(:validate_three_ds_params, post[:threeDSecure])
+
+    assert_equal 'ThreeDs data is invalid', resp.message
+    assert_equal 'ThreeDs version not supported', resp.params['three_ds_version']
+  end
+
+  def test_three_ds_auth_response_validation
+    post = {}
+    @options[:three_ds_options] = @three_ds_secure
+    @gateway.send(:add_three_ds, post, @options)
+    post[:threeDSecure][:external][:status] = 'P'
+    resp = @gateway.send(:validate_three_ds_params, post[:threeDSecure][:external])
+    assert_equal 'ThreeDs data is invalid', resp.message
+    assert_equal 'Authentication response value not supported', resp.params['auth_response']
+  end
+
+  def test_purchase_with_three_ds
+    @options[:three_ds_options] = @three_ds_secure
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
```

Comment:
💬  Comment:

to avoid the unnecessary `respond_with` at the end, use the skip_response param for the check_request method

```Ruby
check_request(skip_response: true)
```